doc: tidy nix, direnv install instructions

ggutoski opened the pull request:
<!---
Credit: Arkworks project https://github.com/arkworks-rs/
-->

<!-- < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < ☺
v                               ✰  Thanks for creating a PR! ✰
v    Before hitting that submit button please review the checkboxes.
v    If a checkbox is n/a - please still include it but + a little note why
☺ > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > >  -->

## Description

<!-- Add a description of the changes that this PR introduces and the files that
are the most critical to review.
-->

A hello-world PR to clarify nix and direnv install instructions in `README.md`.

### TODO

- "git hooks" section: is this necessary? Perhaps it should be moved to a later point in the doc so as to avoid cluttering the basic instructions 
---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (main)
- [x] Linked to GitHub issue with discussion and accepted design OR have an explanation in the PR that describes this work.
- [ ] Wrote unit tests -> doc-only PR
- [ ] Updated relevant documentation in the code -> doc-only PR
- [ ] Added a relevant changelog entry to the `Pending` section in `CHANGELOG.md` -> doc-only PR
- [x] Re-reviewed `Files changed` in the GitHub PR explorer

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/EspressoSystems/jellyfish?pullRequest=181) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/EspressoSystems/jellyfish?pullRequest=181) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/EspressoSystems/jellyfish?pullRequest=181) it.</sub>

mrain reviewed (dismissed):
LGTM. Let's see if @alxiong or @tessico would have more comments.

tessico commented:
Looks good, though we should address the TODOs before merging. @ggutoski what's the reasoning behind the TODO re pre-commit hooks? Is it because these would be run anyway if you commit?

ggutoski commented:
> what's the reasoning behind the TODO re pre-commit hooks?

@tessico The `pre-hooks` command is presented in the "critical path" at the top of the README for setting up the repo, yet I was able to skip this command and run tests etc. without issue. This suggests to me that the `pre-hooks` should not be presented in the critical path.

In [d8e1832](https://github.com/EspressoSystems/jellyfish/pull/181/commits/d8e183284603a9db6f7b02f2eee45017574c1356) I removed the TODO and moved the pre-commit hooks content to the end of the README. I think it's ready to merge.

Other notes to address later:
- IMO the README needs more clean-up but I don't want to turn this quick-and-dirty PR into a time sink.
- Is there a way to bypass CI checks for simple doc PRs such as this? (I tried putting "[no-ci]" in the commit msg 🤷.) It's nice to avoid wasting github compute time on trivial commits.
